recompute_heap.go

```diff
@@ -98,25 +98,24 @@ func (rh *recomputeHeap) removeMinHeight() (nodes []INode) {
 		}
 	}
 
-	if heightBlock == nil {
-		panic(fmt.Errorf("recomputeHeap removeMinHeight block unset or empty with numItems=%d", rh.numItems))
-	}
-
 	heightLen := heightBlock.len()
-	nodes = make([]INode, 0, heightLen)
-	heightBlock.consume(func(id Identifier, n INode) {
+	(*dst) = make([]INode, 0, heightLen)
```

Comment:
note this is basically the consume method on the recompute height list but unrolled to avoid having to work through a callback